doc/debuginfo.txt

```diff
@@ -0,0 +1,187 @@
+Babeltrace Debug Info Analysis
+-----------------------------
+
+The babeltrace debug info analysis is a set of new features which
+allow mapping events from a trace to their location in source code.
+
+Prerequisites
+-------------
+
+In order to install a version of babeltrace with debug info support,
+the following software is required:
+
+    * libelf
+    * libdw
+
+The two libraries required are provided by the elfutils project
+(https://fedorahosted.org/elfutils/), and can be installed through the
+elfutils package on Ubuntu, Debian, RHEL, and others.
+
+Getting Babeltrace with Debug Info Support
+------------------------------------------
+
+The debug info feature branch of babeltrace can be found at
+https://github.com/abusque/babeltrace/tree/debuginfo
+
+The installation procedure is unchanged from other versions, and can
+be found in the README file.
+
+Compiling for Debug Info Analysis
+---------------------------------
+
+Traced programs on which debug info analysis is to be performed can be
+compiled in a few different ways, and still lead to useful results.
+
+Ideally, one should compile the program in debug mode, using simply
+the `-g` flag. This generates DWARF information, which is then used by
```

Comment:
"produces debugging information in the operating system's native format".
We should mention we only support DWARF 2